src/opm/input/eclipse/Schedule/Network/ExtNetwork.cpp

```diff
@@ -75,11 +75,13 @@ const Node& ExtNetwork::root() const {
 
 void ExtNetwork::add_branch(Branch branch)
 {
-    if (!this->has_indexed_node_name(branch.downtree_node())) {
-        this->add_indexed_node_name(branch.downtree_node());
-    }
-    if (!this->has_indexed_node_name(branch.uptree_node())) {
-        this->add_indexed_node_name(branch.uptree_node());
+    for (const std::string& nodename : { branch.downtree_node(), branch.uptree_node() }) {
+        if (!this->has_node(nodename)) {
+            this->m_nodes.insert_or_assign(nodename, Node{nodename});
```

Comment:
It **probably** doesn't matter, but it's slightly faster to construct the `Node` in place within `m_nodes` than to construct a temporary and then moving it into the `m_nodes` container.  I.e.,
```C++
this->m_nodes.emplace(std::piecewise_construct, std::forward_as_tuple(nodename), std::forward_as_tuple(nodename));
```

Reply:
Point taken. I must admit I find it quite hard to read though, and hope it is ok to reserve this for more performance sensitive parts. (This insertion will only be executed max twice per BRANPROP record.)